src/library_webgl.js

```diff
@@ -1454,7 +1454,13 @@ var LibraryGL = {
       return;
     }
 #endif
-    GLctx.texImage2D(target, level, internalFormat, width, height, border, format, type, pixels ? emscriptenWebGLGetTexPixelData(type, format, width, height, pixels, internalFormat) : null);
+    var pixelData = null;
+    if (pixels) {
+      pixelData = emscriptenWebGLGetTexPixelData(type, format, width, height, pixels, internalFormat);
+      if (!pixelData)
```

Comment:
need `{ }` if the if body is on another line (or, for a short one like this, can return on the same line)